### PR TITLE
Make zaaktype fields optional — [#1723]

### LINKED ISSUE #1723: Make setting a zaakstatus optional
### Thema / Theme

Admin

### Omschrijving / Description

Currently, our pattern to create a Zaak is to also set the Zaakstatus. This is configured with mandatory fields in the form registration tab for StUF ZDS.

We need to adjust the react+model to make this optional. If not filled in, don't include the status in the StUF ZDS template (while at it, make zaaktype description optional as well - so only code is required)

We need the same for ZGW APIs but there it's a separate API call to status that needs to be done or not.

### Added value / Toegevoegde waarde

To accommodate Roxit

### Aanvullende opmerkingen / Additional context

_No response_

### PR DESCRIPTION
**Pull request by dariomory**
Fixes #1723

### CODE REVIEW

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
The options form has changed, but I'm not seeing any tests or changes to the StUF-ZDS plugin code to actually deal with the situation where no values are provided for these fields.

The ticket describes what should or shouldn't happen in the StUF-ZDS registration.

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: src/openforms/registrations/contrib/stuf_zds/plugin.py [hunk 1]
```diff
@@ -37,16 +37,19 @@ class ZaakOptionsSerializer(JsonSchemaSerializerMixin, serializers.Serializer):
...
     zds_zaaktype_status_code = serializers.CharField(
-        required=True,
+        required=False,
+        allow_null=True,
         help_text=_("Zaaktype status code for newly created zaken in StUF-ZDS"),
     )
     zds_zaaktype_status_omschrijving = serializers.CharField(
-        required=True,
+        required=False,
+        allow_null=True,
```
Comment (sergei-maertens) [inline src/openforms/registrations/contrib/stuf_zds/plugin.py:52]: there is no reason to set `allow_null=True` - string-based inputs should only have one type of empty value: the empty string.

You need to update the test to use empty strings instead to reflect actual user input.

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: src/openforms/registrations/contrib/stuf_zds/plugin.py (hunk 1 above)
Comment (dariomory) [reply src/openforms/registrations/contrib/stuf_zds/plugin.py:52]: @sergei-maertens The API sends null values in the payload:

image

Comment (sergei-maertens) [reply src/openforms/registrations/contrib/stuf_zds/plugin.py:52]: it should be empty strings instead of null

File: src/stuf/stuf_zds/templates/stuf_zds/soap/creeerZaak.xml [hunk 2]
```diff
@@ -86,8 +90,12 @@
...
         <ZKN:heeft StUF:entiteittype="ZAKSTT" StUF:verwerkingssoort="T">
             <ZKN:gerelateerde StUF:entiteittype="STT" StUF:verwerkingssoort="I">
-                <ZKN:code>{{ zds_zaaktype_status_code }}</ZKN:code>
-                <ZKN:omschrijving>{{ zds_zaaktype_status_omschrijving }}</ZKN:omschrijving>
+                {% if zds_zaaktype_status_code %}
+                    <ZKN:code>{{ zds_zaaktype_status_code }}</ZKN:code>
+                {% endif %}
+                {% if zds_zaaktype_status_omschrijving %}
+                    <ZKN:omschrijving>{{ zds_zaaktype_status_omschrijving }}</ZKN:omschrijving>
+                {% endif %}
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/stuf/stuf_zds/templates/stuf_zds/soap/creeerZaak.xml:98]: clarified - it is!

File: src/openforms/registrations/contrib/stuf_zds/tests/test_backend.py [hunk 3]
```diff
@@ -645,6 +660,416 @@ def test_retried_registration_with_internal_reference(self, m, mock_task):
...
+        self.assertEqual(
+            TimelineLogProxy.objects.filter(
+                template="logging/events/stuf_zds_success_response.txt"
+            ).count(),
+            6,
+        )
+
+    @patch("celery.app.task.Task.request")
+    def test_plugin_optional_fields_missing_status_description(self, m, mock_task):
+        submission = SubmissionFactory.from_components(
```
Comment (sergei-maertens) [inline src/openforms/registrations/contrib/stuf_zds/tests/test_backend.py:869]: I'm missing the variant where `zds_zaaktype_status_omschrijving` is set but `zds_zaaktype_status_code` isn't

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

**Review by sergei-maertens — APPROVED** [review]

**Comment by sergei-maertens** [comment]
Backported in b45116d09908784cf7504766aeb58e7003cb820e, 5a736729a56f5caf97ac753bd4edf45e5f1f54d4 for stable/1.0.x and ea4d386225ffece8c81a2c74dc3103a9501a4249, e7dfbb3cbd938048cf83de0467d0da6bee47792a for stable/1.1.x
